### PR TITLE
 add go1.20 image to build

### PR DESCRIPTION
**Pull request by ArnaudBes**
Create Golang 1.20 image 
Update Changelog 
Update README with tips for local testing

### CODE REVIEW

**Comment by ArnaudBes** [comment]
For reviewer @Bladrak

File: golang/1.20/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+FROM golang:1.20-rc as base
```
Comment (Bladrak) [inline golang/1.20/Dockerfile:1]: Can't we template that? To avoid having one Dockerfile for each version

**Review by Bladrak — CHANGES_REQUESTED** [review]

File: golang/Dockerfile.template [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.18.9 as base
+FROM golang:${GOLANG_VERSION} as base
```
Comment (Bladrak) [inline golang/Dockerfile.template:1]: You could use an ARG, avoids using a script :)

Comment (axelpavageau) [reply golang/Dockerfile.template:1]: the problem with these methods is that dependabot won't be able to create the pull requests to update the dockerfiles.

**Review by Bladrak — CHANGES_REQUESTED** [review]

File: golang/config.yml [hunk 3]
```diff
@@ -9,23 +9,11 @@ test_config: &test_config
     - modd --version
     - rsync --version
     - swagger version
-
+before_script:
```
Comment (Bladrak) [inline golang/config.yml:12]: Still needed?

**Review by Bladrak — CHANGES_REQUESTED** [review]

**Review by Bladrak — APPROVED** [review]

**Review by Bladrak — APPROVED** [review]

**Comment by ArnaudBes** [comment]
replace with new pull request https://github.com/ekino/docker-buildbox/pull/1059

**Pull request closed without merging**